### ISSUE TITLE
`files/eni-max-pods.txt` c5.4xlarge has wrong value

### ISSUE THREAD

**Issue by iaroslav-ciupin** [awslabs/amazon-eks-ami#1368]
**What happened**:
`files/eni-max-pods.txt` file for `c5.4xlarge` instance has value 234 (same as for c5.9xlarge).

But max-pods-calculator.sh gives different result. 
```bash
> ./max-pods-calculator.sh --instance-type c5.4xlarge --cni-version 1.11.4-eksbuild.1 
110
```

Neighbour values seems to be correct:
c5.2xlarge has 58
c5.9xlarge has 234

**What you expected to happen**:
In `files/eni-max-pods.txt` `c5.4xlarge` instance most probably should have 110 (not 234).

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

Is this file auto-generated or manually maintained by developers? how to properly fix this: just adjust the file or fix the code-generation script(if so, where's the script)?

**Comment by bryantbiggs** [comment]
This is the source of truth https://github.com/aws/amazon-vpc-cni-k8s/blob/master/scripts/gen_vpc_ip_limits.go 

However, the number is correct. The max pods listed in `eni-max-pods.txt` is a theoretical limit based solely on the networking constructs provided by the VPC networking (number of ENIs and number of IPs per ENI, etc.). The max pods calculator has additional considerations from a resource perspective. For this scenario, since c5.4xlarge has less than 30 vCPUs, the limit is capped at 110 when provisioning the node https://github.com/awslabs/amazon-eks-ami/blob/55cbc55f69a029dc821d16081a7c75c206a5d1cd/files/max-pods-calculator.sh#L155

tl;dr - `max-pods-calcualtor.sh` results do not reflect the values of `eni-max-pods.txt`

**Comment by iaroslav-ciupin** [comment]
@bryantbiggs I don't get why there is a discrepancy between value from `eni-max-pods.txt` and value provided by calculator. Shouldn't they provide same values? or are they serving different purposes and should be used in different contexts?

**Comment by bryantbiggs** [comment]
The ENI max pods is what it says on the tin - the maximum number of pods that the instance could support strictly from an ENI perspective (this is a theoretical limit, for some instances it will not be practical, like the c5d.18xlarge which theoretically could support 737 pods but this would be quite a resource constrained setup). Said differently, think of the ENI max pods as the theoretical maximum number of IPs the instance could support

The max pods calculator combines several different factors, one of which is the overhead of pod resources, to calculate the maximum number of pods the instance could support. So, even if the instance has the ability to provide 737 pods, there wouldn't be enough resources to sustain that many pods and therefore the number is capped much lower

**Comment by bryantbiggs** [comment]
Also, you can completely bypass the max pods calculator and run more pods than the 110/250 limit imposed by the calculator. However, you cannot run more pods than what is calculated by the ENI max pods unless you use a different form of networking (overlay networking, etc.) other than the AWS VPC networking

**Comment by iaroslav-ciupin** [comment]
@bryantbiggs thank you for explaining!
